### ISSUE TITLE
Error: Cannot read property 'value' of undefined

### ISSUE THREAD

**Issue by erin-ntnu** [giosans/multiregional-landslide-detection-article#2]
Hi! 

After changing out the google fusion table for a .csv file, now I have a problem where it is bouncing back and forth between two lines where it is trying to calculate the end date. 

I can't spot what the problem is.  As far as I can see, all the variables and functions in that segment have been used previously in the code without getting that error. Are you able to see it?  

https://code.earthengine.google.com/?scriptPath=users%2Ferinpilaar%2FPreprocessing_EL%3APreprocessing-EL

![image](https://user-images.githubusercontent.com/94454477/142617958-395e1fef-85fd-4b01-8136-f21b1633e7a4.png)

![image](https://user-images.githubusercontent.com/94454477/142618326-ff5c6f6d-5fbf-47d7-a7dd-1564ccae4d72.png)

**Comment by giosans** [comment]
Hey! May have to make it public cause I can't see it. It seems those variables don't have value property, so I'd debug the code by inspecting those ev_result and end_date_result and check what the properties are.

**Comment by erin-ntnu** [comment]
git clone https://earthengine.googlesource.com/users/erinpilaar/Preprocessing_EL

Hi! can you open this?

**Comment by erin-ntnu** [comment]
https://code.earthengine.google.com/?accept_repo=users/erinpilaar/Preprocessing_EL 

Hi! Does this one work, I have allowed read access for all.

**Comment by giosans** [comment]
hey! It does work, though it's blank.

**Comment by erin-ntnu** [comment]
Uff.... Is there nothing under the Scripts tab?
![image](https://user-images.githubusercontent.com/94454477/143036371-ce3bbdc1-f305-4b2f-ab59-bc68288f00d2.png)

**Comment by giosans** [comment]
Hihi! Cant see your users/erinpillar/ folder at all. If not in the writer folder, I should at least see it in the reader folder. Cant remember how to share resources, sorry. This article may help you https://medium.com/google-earth/making-it-easier-to-reuse-code-with-earth-engine-script-modules-2e93f49abb13

**Comment by erin-ntnu** [comment]
Hey! 

So for anyone else trying- I have made it run finally! 

The problems were mainly to do with the formatting of the date and coordinates in the csv that I was trying to use instead of a Goggle Fusion table. 
My solution was to use a shapefile, I created one in ArcMap, with a date column 'event_date' written in the format yyyy-MM-dd'T'HH:mm:ss.SSSZ, e.g. 2019-07-30T00:00:00.000+0100. 

There was also an issue with the ALOS_dem. I have a feeling that originally there was only one band in that dataset and now there are multiple. There were problems to reproject it because it was still behaving as an 'Image Collection', rather than as an Image. My solution was to upload my own DEM with a single band as a geotiff asset, and change the code references accordingly.
